staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/encode.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package modes
+
+import (
+	"github.com/fxamacker/cbor/v2"
+)
+
+var Encode cbor.EncMode = func() cbor.EncMode {
+	encode, err := cbor.EncOptions{
+		// Map keys need to be sorted to have deterministic output, and this is the order
+		// defined in RFC 8949 4.2.1 "Core Deterministic Encoding Requirements".
+		Sort: cbor.SortBytewiseLexical,
+
+		// CBOR supports distinct types for IEEE-754 float16, float32, and float64. Store
+		// floats in the smallest width that preserves value so that equivalent float32 and
+		// float64 values encode to identical bytes, as they do in a JSON
+		// encoding. Satisfies one of the "Core Deterministic Encoding Requirements".
+		ShortestFloat: cbor.ShortestFloat16,
+		NaNConvert:    cbor.NaNConvert7e00,
+		InfConvert:    cbor.InfConvertFloat16,
```

Comment:
these don't look like sizes, they look like handling for invalid inputs?  What do these do?

Reply:
The ShortestFloatMode doesn't apply to Inf/NaN values. InfConvertFloat16 encodes +Inf and -Inf to float16. There are a bunch of representations of NaN in IEEE 754, but the CBOR spec recommends that you pick a single representation if your application does not smuggle extra information inside NaN values. NaNConvert7e00 always uses the float16 0x7e00 representation.

There's an item on the tracker issue to decide whether to reject NaN/Inf values entirely. JSON can't represent them, Protobuf does.

Reply:
Added more explanatory comments for the Inf/Nan options.